odd-platform-api/src/main/java/org/opendatadiscovery/oddplatform/service/IngestionServiceImpl.java

```diff
@@ -101,23 +107,62 @@ public Mono<Void> ingest(final DataEntityList dataEntityList) {
                     .stream()
                     .filter(dto -> BooleanUtils.isTrue(dto.getDatasetSchemaChanged()))
                     .map(EnrichedDataEntityIngestionDto::getId)
-                    .collect(Collectors.toList());
-
-                final Map<String, DatasetStructureDelta> datasetStructureDelta =
-                    datasetVersionRepository.getLastStructureDelta(changedSchemaIds);
-
+                    .toList();
+
+                return reactiveDatasetVersionRepository.getLatestVersions(changedSchemaIds)
+                    .flatMap(latestVersions -> {
+                        final List<DatasetVersionPojo> latestVersionsWithPenultimates = latestVersions.stream()
+                            .filter(p -> p.getVersion() > 1)
+                            .collect(Collectors.toList());
+                        return reactiveDatasetVersionRepository.getPenultimateVersions(latestVersionsWithPenultimates)
```

Comment:
I think it's better to write the next code before return statement

```
if (latestVersionsWithPenultimates.isEmpty()) {
    return Mono.just(Map.of(), dataStructure);
}
```

In this case return statement won't be executed, which is expected when we don't have penultimate versions